### PR TITLE
fix(util): fix the problem of pod overhead not being calculated

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
- fix the problem of overhead not being calculated
#### Which issue(s) this PR fixes:

Fixes None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/util/resource.go [hunk 1]
```diff
@@ -73,5 +74,14 @@ func GetPodEffectiveRequest(pod *v1.Pod) v1.ResourceList {
 		}
 		resources[name] = quantity
 	}
+	// Add pod overhead if present
+	if pod.Spec.Overhead != nil {
```
Comment (Huang-Wei) [inline pkg/util/resource.go:78]: Overhead is a map, so we can directly iterate it w/o this if check.

File: pkg/util/resource_test.go [hunk 2]
```diff
@@ -119,6 +141,9 @@ func TestGetPodEffectiveRequest(t *testing.T) {
 					},
 				})
 			}
+			if tt.podOverheadRequest != nil {
```
Comment (Huang-Wei) [inline pkg/util/resource_test.go:144]: ditto. We can assign it directly w/o the if check.

**Review by Huang-Wei — COMMENTED** [review]
A nit. LGTM otherwise.

File: pkg/util/resource_test.go (hunk 2 above)
Comment (googs1025) [reply pkg/util/resource_test.go]: done

File: pkg/util/resource.go (hunk 1 above)
Comment (googs1025) [reply pkg/util/resource.go]: done

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**